src/api/vfs/sync_io.rs

```diff
@@ -116,6 +118,16 @@ impl FileSystem for Vfs {
                 fs.getattr(ctx, idata.ino(), handle)
                     .map(|(mut attr, duration)| {
                         attr.st_ino = idata.into();
+                        // If id_mapping is enabled, map the internal ID to the external ID.
+                        if let Some(id_mapping) = self.id_mapping {
+                            let (internal_id, external_id, range) = id_mapping;
```

Comment:
Please add a helper function to handle id remapping for attr, and use it to avoid duplicating the same code in multiple places.

Reply:
Sure, thanks! I'll get right on that and make the changes.